WordPress/src/test/java/org/wordpress/android/ui/stories/SaveStoryGutenbergBlockUseCaseTest.kt

```diff
@@ -0,0 +1,288 @@
+        Assertions.assertThat(mediaFileData.alt).isEqualTo("")
+        Assertions.assertThat(mediaFileData.id).isEqualTo(TEMPORARY_ID_PREFIX + 1)
+        Assertions.assertThat(mediaFileData.link).isEqualTo(
+                "https://testsite.files.wordpress.com/2020/10/wp-0000000.jpg"
+        )
+        Assertions.assertThat(mediaFileData.type).isEqualTo("image")
+        Assertions.assertThat(mediaFileData.mime).isEqualTo(mediaFile.mimeType)
+        Assertions.assertThat(mediaFileData.caption).isEqualTo("")
+        Assertions.assertThat(mediaFileData.url).isEqualTo(mediaFile.fileURL)
+    }
+
+    @Test
+    fun `local media id is found and gets replaced with remote media id`() {
+        // Given
+        val mediaFile = getOneMediaFile(1)
+        val onePost = PostModel()
+        onePost.setContent(blockWithNonEmptyMediaFiles)
+
+        // When
+        saveStoryGutenbergBlockUseCase.replaceLocalMediaIdsWithRemoteMediaIdsInPost(
+                onePost,
+                mediaFile
+        )
+
+        // Then
+        Assertions.assertThat(onePost.content).isEqualTo(blockWithNonEmptyMediaFilesWithOneRemoteId)
+    }
+
+    @Test
+    fun `slides are saved locally to storiedPrefs`() {
+        // Given
+        val frames = ArrayList<StoryFrameItem>()
+        frames.add(getOneStoryFrameItem("1"))
+        frames.add(getOneStoryFrameItem("2"))
+        frames.add(getOneStoryFrameItem("3"))
+
+        // When
+        saveStoryGutenbergBlockUseCase.saveNewLocalFilesToStoriesPrefsTempSlides(
+                siteModel,
+                0,
+                frames
+        )
+
+        // Then
+        verify(storiesPrefs, times(3)).saveSlideWithTempId(any(), any(), any())
+    }
+
+    private fun setupFluxCMediaFiles(
+        emptyList: Boolean
+    ): ArrayList<MediaFile> {
+        when (emptyList) {
+            true -> return ArrayList()
+            false -> {
+                val mediaFiles = ArrayList<MediaFile>()
+                for (i in 1..10) {
+                    val oneMediaFile = MediaFile()
+                    oneMediaFile.id = i
+                    oneMediaFile.mediaId = (i + 1000).toString()
+                    oneMediaFile.mimeType = "image/jpeg"
+                    oneMediaFile.fileURL = "https://testsite.files.wordpress.com/2020/10/wp-0000000.jpg"
+                    mediaFiles.add(oneMediaFile)
+                }
+                return mediaFiles
+            }
+        }
+    }
+
+    private fun getOneMediaFile(id: Int): MediaFile {
+        val oneMediaFile = MediaFile()
+        oneMediaFile.id = id
+        oneMediaFile.mediaId = (id + 1000).toString()
+        oneMediaFile.mimeType = "image/jpeg"
+        oneMediaFile.fileURL = "https://testsite.files.wordpress.com/2020/10/wp-0000000.jpg"
+        return oneMediaFile
+    }
+
+    private fun getOneStoryFrameItem(id: String): StoryFrameItem {
+        return StoryFrameItem(
+                source = mock(),
+                id = id
+        )
+    }
```

Comment:
Could we remove the `one` in the names of the functions/variables here? If not then let me know. Thanks :)

Reply:
👍 done 87d8f7b4e1

Reply:
Thanks. Could we do the method names as well?

Reply:
sure, done in f324acdce3